### PR TITLE
Add triangulated flag to Mesh.to_vertices_and_faces

### PR DESCRIPTION
**Pull request by gonzalocasas**
For consistency with the other `to_vertices_and_faces` methods in all shapes, I added the same `triangulated=False` optional flag to `Mesh` (and also to the abstract method of `Shape`)

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/datastructures/mesh/mesh.py [hunk 1]
```diff
@@ -486,7 +491,36 @@ def to_vertices_and_faces(self):
...
+                vertices.append(centroid)
+                vcount = len(vertices)
+
+                for i in range(len(face_vertices) - 1):
+                    a, b, c = face_vertices[i], face_vertices[i + 1], vcount - 1
+                    faces.append([key_index[a], key_index[b], c])
+
+                # last face between first, last and centroid
+                a, b, c = face_vertices[0], face_vertices[-1], vcount - 1
+                faces.append([key_index[b], key_index[a], c])
```
Comment (tomvanmele) [inline src/compas/datastructures/mesh/mesh.py]: this is what pairwise is for
```python
for a, b in pairwise(face_vertices + face_vertices[:1]):
    faces.append([key_index[a], key_index[b], vcount - 1])
```

also, if you do `len(vertices)` before appending the centroid, you can avoid the repeated `vcount - 1`

File: src/compas/datastructures/mesh/mesh.py [hunk 2]
```diff
@@ -486,7 +491,36 @@ def to_vertices_and_faces(self):
...
+        for fkey in self.faces():
+            face_vertices = self.face_vertices(fkey)
+
+            if len(face_vertices) == 3:
+                a, b, c = face_vertices
+                faces.append([key_index[a], key_index[b], key_index[c]])
+            elif len(face_vertices) == 4:
+                a, b, c, d = face_vertices
+                faces.append([key_index[a], key_index[b], key_index[c]])
+                faces.append([key_index[a], key_index[c], key_index[d]])
```
Comment (tomvanmele) [inline src/compas/datastructures/mesh/mesh.py:510]: her we could potentially check for the "better" option.
but perhaps that is for another PR :)

File: src/compas/datastructures/mesh/mesh.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas/datastructures/mesh/mesh.py]: fixed at https://github.com/compas-dev/compas/pull/1013/commits/dd3f87fc64a4038874a51cfd8a95ee738ed318bd

**Comment by gonzalocasas** [comment]
also removed additional lookup of vertex coordinates at https://github.com/compas-dev/compas/pull/1013/commits/d617fb8408c7db62c9a0ba4926eed320a932e898

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**